src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,251 @@
+	 *                                             $version is set to false, the version number is the currently installed
+	 *                                             WordPress version. If $version is set to null, no version is added.
+	 */
+	public function register( $module_identifier, $src, $dependencies = array(), $version = false ) {
+		if ( ! isset( $this->registered[ $module_identifier ] ) ) {
+			$deps = array();
+			foreach ( $dependencies as $dependency ) {
+				if ( isset( $dependency['id'] ) ) {
+					$deps[] = array(
+						'id'   => $dependency['id'],
+						'type' => isset( $dependency['type'] ) && 'dynamic' === $dependency['type'] ? 'dynamic' : 'static',
+					);
+				} elseif ( is_string( $dependency ) ) {
+					$deps[] = array(
+						'id'   => $dependency,
+						'type' => 'static',
+					);
+				}
+			}
+
+			$this->registered[ $module_identifier ] = array(
+				'src'          => $src,
+				'version'      => $version,
+				'enqueued'     => isset( $this->enqueued_before_registered[ $module_identifier ] ),
+				'dependencies' => $deps,
+			);
+		}
+	}
+
+	/**
+	 * Marks the module to be enqueued in the page.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public function enqueue( $module_identifier ) {
+		if ( isset( $this->registered[ $module_identifier ] ) ) {
+			$this->registered[ $module_identifier ]['enqueued'] = true;
+		} else {
+			$this->enqueued_before_registered[ $module_identifier ] = true;
+		}
+	}
+
+	/**
+	 * Unmarks the module so it is no longer enqueued in the page.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public function dequeue( $module_identifier ) {
+		if ( isset( $this->registered[ $module_identifier ] ) ) {
+			$this->registered[ $module_identifier ]['enqueued'] = false;
+		}
+		unset( $this->enqueued_before_registered[ $module_identifier ] );
+	}
+
+	/**
+	 * Returns the import map array.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array Array with an `imports` key mapping to an array of module identifiers and their respective URLs,
+	 *               including the version query.
+	 */
+	public function get_import_map() {
+		$imports = array();
+		foreach ( $this->get_dependencies( array_keys( $this->get_enqueued() ) ) as $module_identifier => $module ) {
+			$imports[ $module_identifier ] = $module['src'] . $this->get_version_query_string( $module['version'] );
+		}
+		return array( 'imports' => $imports );
+	}
+
+	/**
+	 * Prints the import map using a script tag with a type="importmap" attribute.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_import_map() {
+		$import_map = $this->get_import_map();
+		if ( ! empty( $import_map['imports'] ) ) {
+			wp_print_inline_script_tag(
+				wp_json_encode( $import_map, JSON_HEX_TAG | JSON_HEX_AMP ),
+				array(
+					'type' => 'importmap',
```

Comment:
Adding an `id` here would seem useful as WP generally adds IDs to other scripts bring printed to the page.

```suggestion
					'type' => 'importmap',
 					'id'   => 'wp-js-module-importmap',

```

Reply:
Done as part of https://github.com/WordPress/wordpress-develop/pull/5818/commits/834693d4a2223a52bb3f78ab410c8a867108aa5d.